devito/finite_differences/differentiable.py

```diff
@@ -730,6 +730,12 @@ def __init_finalize__(self, *args, **kwargs):
 
         super().__init_finalize__(*args, **kwargs)
 
+    @classmethod
+    def class_key(cls):
+        # Ensure Weights appear before any other AbstractFunction
```

Comment:
Why is this needed? Genuinely curious

Reply:
sympy voodoo, sigh

to avoid that e.g. CUDA unvectorized code looks like `a[i]*b[i]` and like `ab[i].y*ab[i].x` once vectorized (ie, with operands flipped). In short, for consistency of the order of arithmetic operations